tests/equality/equality.cpp

```diff
@@ -0,0 +1,995 @@
+
+    r1->setVariable(v1);
+    r2->setVariable(v2);
+
+    EXPECT_FALSE(r1->equal(r2));
+}
+
+TEST(Equality, resetNotEqualByTestVariable)
+{
+    libcellml::ResetPtr r1 = libcellml::Reset::create();
+    libcellml::ResetPtr r2 = libcellml::Reset::create();
+
+    libcellml::VariablePtr v1 = libcellml::Variable::create("vA");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("vA");
+
+    libcellml::VariablePtr tv1 = libcellml::Variable::create("vTestA");
+    libcellml::VariablePtr tv2 = libcellml::Variable::create("vTest");
+
+    r1->setOrder(3);
+    r2->setOrder(3);
+
+    r1->setResetValue("reset value math");
+    r2->setResetValue("reset value math");
+
+    r1->setResetValueId("rv_id");
+    r2->setResetValueId("rv_id");
+
+    r1->setTestValue("test value math");
+    r2->setTestValue("test value math");
+
+    r1->setTestValueId("tv_id");
+    r2->setTestValueId("tv_id");
+
+    r1->setTestVariable(tv1);
+    r2->setTestVariable(tv2);
+
+    r1->setVariable(v1);
+    r2->setVariable(v2);
+
+    EXPECT_FALSE(r1->equal(r2));
+}
+
+TEST(Equality, resetNotEqualByVariable)
+{
+    libcellml::ResetPtr r1 = libcellml::Reset::create();
+    libcellml::ResetPtr r2 = libcellml::Reset::create();
+
+    libcellml::VariablePtr v1 = libcellml::Variable::create("vA");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("vB");
+
+    libcellml::VariablePtr tv1 = libcellml::Variable::create("vTest");
+    libcellml::VariablePtr tv2 = libcellml::Variable::create("vTest");
+
+    r1->setOrder(3);
+    r2->setOrder(3);
+
+    r1->setResetValue("reset value math");
+    r2->setResetValue("reset value math");
+
+    r1->setResetValueId("rv_id");
+    r2->setResetValueId("rv_id");
+
+    r1->setTestValue("test value math");
+    r2->setTestValue("test value math");
+
+    r1->setTestValueId("tv_id");
+    r2->setTestValueId("tv_id");
+
+    r1->setTestVariable(tv1);
+    r2->setTestVariable(tv2);
+
+    r1->setVariable(v1);
+    r2->setVariable(v2);
+
+    EXPECT_FALSE(r1->equal(r2));
+}
+
+TEST(Equality, resetNotEqualByNullTestVariable)
+{
+    libcellml::ResetPtr r1 = libcellml::Reset::create();
+    libcellml::ResetPtr r2 = libcellml::Reset::create();
+
+    libcellml::VariablePtr v1 = libcellml::Variable::create("vA");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("vB");
+
+    libcellml::VariablePtr tv1 = libcellml::Variable::create("vTest");
+    libcellml::VariablePtr tv2 = libcellml::Variable::create("vTest");
+
+    r1->setOrder(3);
+    r2->setOrder(3);
+
+    r1->setResetValue("reset value math");
+    r2->setResetValue("reset value math");
+
+    r1->setResetValueId("rv_id");
+    r2->setResetValueId("rv_id");
+
+    r1->setTestValue("test value math");
+    r2->setTestValue("test value math");
+
+    r1->setTestValueId("tv_id");
+    r2->setTestValueId("tv_id");
+
+    r2->setTestVariable(tv2);
+
+    r1->setVariable(v1);
+    r2->setVariable(v2);
+
+    EXPECT_FALSE(r1->equal(r2));
```

Comment:
Should all these tests also include the reverse test too for completeness?  ie: `r2->equal(r1)` ?

Reply:
82ed235